src/Controllers/ViewerController.cs

```diff
@@ -7,9 +10,34 @@ namespace GroupDocs.Viewer.MVC.Controllers
     /// </summary>
     public class ViewerController : Controller
     {
-        public ActionResult Index()
+        public ActionResult Index(string file)
         {
-            return this.View();
+            if (!string.IsNullOrEmpty(file))
+            {
+                string documentGuid = file;
+
+                Uri uriResult;
+                bool result = Uri.TryCreate(file, UriKind.Absolute, out uriResult)
+                    && (uriResult.Scheme == Uri.UriSchemeHttp || uriResult.Scheme == Uri.UriSchemeHttps);
+
+                if (result)
+                {
+                    string fileSavePath = DocumentDataLoader.GetFileSavePath(file);
+                    documentGuid = fileSavePath;
+                }
+
+                var postedData = new PostedDataEntity()

```

Comment:
Codacy found an issue: [Remove these redundant parentheses.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5958475)